_posts/2020-06-07-higher-order-hydras.markdown

```diff
@@ -0,0 +1,491 @@
+
+![Diagram of a Buchholz hydra with the labels drawn to the left instead of on the nodes.](/assets/hydras/externalize-0.png)
+
+We're first going to mark all existing edges with label \\\(0\\\).
+
+![Diagram of a Buchholz hydra with the labels drawn to the left instead of on the nodes, and edges labelled with 0.](/assets/hydras/externalize-1.png)
+
+We're then going to move the current label into a child node, with edge labelled \\\(1\\\).
+
+![Diagram of a Buchholz hydra with the former labels drawn as children with edge label 1, and original edges labelled with 0.](/assets/hydras/externalize-2.png)
+
+All original labels have now been evicted.
+We transfer these edge labels into the now unlabelled children.
+The root could be considered to have a special fake label, or no label.
+
+![Diagram of a Buchholz hydra with the former labels drawn as children with label 1, and original nodes labelled 0.](/assets/hydras/externalize-3.png)
+
+We then expand these former labels into the appropriate hydras, except that the immediate children of that root has label \\\(1\\\), and then fuse those with their parents.
+Recall that the lone root hydra is ordinal \\\(0\\\), a hydra with \\\(n\\\) children is the ordinal \\\(n\\\), and the hydra \\\(((()))\\\) is the ordinal \\\(\omega\\\).
+You'll notice in this step since \\\(0\\\) turned into \\\(()\\\) and got fused with its parent, that branch just disappears, and since \\\(2\\\) turned into \\\((()())\\\) and got fused with its parent, it gained another child with label \\\(1\\\).
+
+![A transformed hydra with only 0 and 1 labels and some new branches.](/assets/hydras/externalize-4.png)
+
+We're almost there!
+We just need to get rid of the remaining ordinal labels by recursively converting them into hydras.
+
+![A transformed hydra with no ordinals, just more hydras. Roots are specially marked with a star symbol.](/assets/hydras/externalize-5.png)
+
+There we have it, our first Amiko hydra!
+Whatever rules we use to define its evolution, it should evolve pretty much the same as Buchholz hydras, for hydras that were converted from Buchholz hydras.
+
+Amiko hydras with only \\\(0 = (\star:)\\\) and \\\(1 = (\star:((\star:):))\\\) labels correspond to nested Buchholz hydras.
+Amiko hydras with labels below \\\(\omega = (\star:(0:(0:))) = (\star:((\star:):((\star:):)))\\\) correspond to hyper-nested Buchholz hydras with a finite number of hyper-nesting dimensions,
+which we may collectively call \\\(\textrm{hyper}^\omega\\\)-nested Buchholz hydras.
+The full construction of Amiko hydras, obtained by applying a close transform after this externalize, is far more powerful,
+and its corresponding ordinal is the first common fixed point of "\\\(\alpha\\\) maps to the ordinal of \\\(\textrm{hyper}^\alpha\\\)-nested Buchholz hydras."
+
+</div>
+
+# Form of Amiko hydras
+{: #amiko-hydras-form}
+
+{% include collapser.markdown %}
+
+<div>
+
+A hydra is an ordered rooted labelled tree.
```

Comment:
This sentence seems to be at odds with a previous sentence from the start: "Kirby-Paris hydras came first. For Kirby-Paris hydras, the rules are simple. _Start with your hydra, which is an_ **unordered unlabelled** _rooted tree_".

Reply:
No, this is intentional. Only a subset of unordered trees (and that subset is all ordered) are in normal form. Kirby-Paris hydras care less and the hydra theorem holds even when the hydras aren't in normal form, but you need to take only the normal form hydras to get an ordering up to e0. Buchholz hydras meanwhile don't have a hydra theorem that I know of that proves it outside of normal form, so they have this same restriction. For my hydras I know for sure normal form is a strict requirement, and I assume normal form in the rules.

Reply:
Oh, I understand now! That makes a lot more sense 👍